backend/src/main/java/org/cryptomator/hub/api/VaultResource.java

```diff
@@ -325,22 +331,43 @@ public VaultDto get(@PathParam("vaultId") UUID vaultId) {
 	@Consumes(MediaType.APPLICATION_JSON)
 	@Produces(MediaType.APPLICATION_JSON)
 	@Transactional
-	@Operation(summary = "creates a vault",
-			description = "Creates a vault with the given vault id. The creationTime in the vaultDto is ignored and the current server time is used.")
-	@APIResponse(responseCode = "201", description = "vault created")
-	@APIResponse(responseCode = "409", description = "vault with given id or name already exists")
-	public Response create(@PathParam("vaultId") UUID vaultId, @Valid VaultDto vaultDto) {
-		if (vaultDto == null) {
-			throw new BadRequestException("Missing vault dto");
+	@Operation(summary = "creates or updates a vault",
+			description = "Creates or updates a vault with the given vault id. The creationTime in the vaultDto is always ignored. On creation the current server time is used. On update, only the name, description and archived fields are considered.")
+	@APIResponse(responseCode = "200", description = "existing vault updated")
```

Comment:
200 or 204 are both fine, I don't have any preference. However I would like all our resources to behave the same way. Have you taken a look at other "update-PUTs"?

see https://stackoverflow.com/a/2342589/4014509

Reply:
We only have one other entity, which gets PUT'ed: Devices in the device resource. And there we only create devices (no update).